zerver/actions/create_user.py

```diff
@@ -368,6 +368,7 @@ def notify_created_bot(user_profile: UserProfile) -> None:
     )
 
 
+@transaction.atomic
```

Comment:
I think we want:
```suggestion
@transaction.atomic(durable=True)
```

We should possibly make a linter for bare `@transaction.atomic` decorators, since we want most of them either be `savepoint=False` _or_ `durable=True`.  But there are a bunch of outstanding bare ones already. 
 Ref #20544 for prior work around this.